metabase/metabase
# fix pivot table horizontal scrollbar not appearing

Closes https://github.com/metabase/metabase/issues/43215

Previously the horizontal scroll bar in the pivot table body was not always visible when it should have been.

The root cause was that the `height` parameter from `react-virtualized`'s `AutoSize` component was providing an incorrect value on the initial page render. It was too large, because it was including the height of the query builder footer area below, so the grid ends up filling that height, and the scrollbar is rendered beneath the footer.

To fix this, we use the `height` prop provided from the `Visualization` component instead, as it provides a more accurate value. Note that we only use this prop in the query builder, because unfortunately it is incorrect for dashcards since it includes the title as well. I tried to fix the `height` prop measurement by providing the right class name for the selector of the `ExplicitSize` component in `Visualization`, but it caused further issues with the height prop decreasing very slowly (our measurements are slow), which caused the scrollbar to remain offscreen for a long time after narrowing the window.

### How to verify

Create the following question and add it to a dashboard. Confirm the scroll bar is visible in both the query builder and in a dashcard, both on initial load and after resizing the window.

<img width="1214" alt="Screenshot 2024-06-20 at 5 41 32 PM" src="https://github.com/metabase/metabase/assets/37751258/61dd8540-bccf-44a6-a613-2e7a9fceb84c">

https://github.com/metabase/metabase/assets/37751258/ee76fba1-c2b5-4b8e-a9bf-c14cf6a81646

Note that on dashboards the scrollbar may be hidden after resizing the dashcard. This is an existing bug, separate from the P1 issue that this PR aims to fix, so we can fix it later.
